lib/utils/SynchronizedQueue.hpp

```diff
@@ -0,0 +1,94 @@
+#ifndef SYNCHRONIZED_QUEUE_HPP
```

Comment:
Needs an `@file` block for doxygen.

Reply:
Done in 345f99ba101a6580d76c70f1f6d03a922eca3352